awx/main/migrations/0118_v380_inventory_plugins.py

```diff
@@ -0,0 +1,83 @@
+# Generated by Django 2.2.11 on 2020-07-20 19:56
+
+import logging
+import json
+
+from django.db import migrations
+
+from ._inventory_source_vars import FrozenInjectors
+
+
+logger = logging.getLogger('awx.main.migrations')
+BACKUP_FILENAME = '/tmp/tower_migration_inventory_source_vars.json'
+
+
+def _get_inventory_sources(InventorySource):
+    # TODO: Maybe pull the list of cloud sources from code
+    return InventorySource.objects.filter(source__in=['ec2', 'gce', 'azure_rm', 'vmware', 'satellite6', 'openstack', 'rhv', 'tower'])
+
+
+def inventory_source_vars_forward(apps, schema_editor):
+    InventorySource = apps.get_model("main", "InventorySource")
+    source_vars_backup = dict()
+
+    for inv_source_obj in _get_inventory_sources(InventorySource):
+        # TODO: Log error if this is false, it shouldn't be false
+        if inv_source_obj.source in FrozenInjectors:
+            source_vars_backup[inv_source_obj.id] = dict(inv_source_obj.source_vars_dict)
+            with open(BACKUP_FILENAME, 'w') as fh:
+                json.dump(source_vars_backup, fh)
+
+            injector = FrozenInjectors[inv_source_obj.source]()
+            new_inv_source_vars = injector.inventory_as_dict(inv_source_obj, None)
+            inv_source_obj.source_vars = new_inv_source_vars
+            inv_source_obj.save()
+
+
+def inventory_source_vars_backward(apps, schema_editor):
```

Comment:
I plan on removing this before merge. It's nice to be able to backward when developing but alas customers almost certainly cram sensitive info into their source_vars